modin/engines/base/frame/data.py

```diff
@@ -1082,32 +1085,21 @@ def _compute_map_reduce_metadata(self, axis, new_parts, preserve_index=True):
         new_axes, new_axes_lengths = [0, 0], [0, 0]
 
         new_axes[axis] = ["__reduced__"]
-        if preserve_index:
-            new_axes[axis ^ 1] = self.axes[axis ^ 1]
-        else:
-            new_axes[axis ^ 1] = self._compute_axis_labels(axis ^ 1, new_parts)
+        new_axes[axis ^ 1] = self.axes[axis ^ 1]
 
         new_axes_lengths[axis] = [1]
         new_axes_lengths[axis ^ 1] = self._axes_lengths[axis ^ 1]
 
-        if axis == 0 or self._dtypes is None:
-            new_dtypes = self._dtypes
-        elif preserve_index:
-            new_dtypes = pandas.Series(
-                [find_common_type(self.dtypes.values)], index=new_axes[axis]
-            )
-        else:
-            new_dtypes = None
+        new_dtypes = None
         result = self.__constructor__(
             new_parts,
             *new_axes,
             *new_axes_lengths,
             new_dtypes,
         )
-        result._apply_index_objs(axis)
```

Comment:
Do we no longer need this because we don't need to explicitly preserve the index?

Reply:
Row/column Index is always preserved for these functionalities.